### PR TITLE
Migrate click attribution tests to Playwright

### PR DESCRIPTION
**Pull request by sammacbeth**
Seperated from #1939 because this migration was a bit more complex.

### CODE REVIEW

File: integration-test/click-attribution.spec.js [hunk 1]
```diff
@@ -39,48 +32,43 @@ describe('Ad click blocking', () => {
...
-        await expectAsync(page.waitForFunction(callback, { timeout: 2000 }, variableName))
-            .withContext(`waitForVariable for '${variableName}'`)
-            .not.toBeRejected()
+        await page.waitForFunction(callback, variableName, { timeout: 2000 })
         return page.evaluate(callback, variableName)
     }
 
     /**
      * Clicks on an item and awaits a new tab to load.
      * https://github.com/puppeteer/puppeteer/issues/3667 without this intercepting new tab requests is too late
```
Comment (jonathanKingston) [inline integration-test/click-attribution.spec.js:41]: Is this issue still relevant?

File: integration-test/click-attribution.spec.js
Comment (jonathanKingston) [inline integration-test/click-attribution.spec.js:1]: There's a `Puppeteer.page` type in this file still (in a comment so not important etc).

File: integration-test/click-attribution.spec.js (hunk 1 above)
Comment (jonathanKingston) [reply integration-test/click-attribution.spec.js:41]: Also hanging brace here.

**Review by jonathanKingston — APPROVED** [review]
Looks good some comment fixing to happen :)

**Comment by sammacbeth** [comment]
Merging this with a failing GPC test. I think this is failing because of config changes I made as part of #1951, and I'll fix it there.

**Pull request merged**